### ISSUE TITLE
Unable to resolve username to process for Windows builtin and elevated accounts

### ISSUE THREAD

**Issue by Rich5** [facebook/osquery#4360]
### What version of `osquery` are you using? What operating system. Be specific.
```
+-----------------+------------+----------+-------------+
| osquery_version | os_version | os_build | os_platform |
+-----------------+------------+----------+-------------+
| 2.11.2          | 6.1.7601   | 7601     | windows     |
+-----------------+------------+----------+-------------+
```

```
OS Name:                   Microsoft Windows 7 Enterprise
OS Version:                6.1.7601 Service Pack 1 Build 7601
```

### What have you already tried?

I'm trying to get similar process details as shown in the native task manager application. Specifically I'd like to show the user account associated with a running process. The following query is the best results I can get so far:

```
select username,uid,path from processes left join users using(uid);
```

### What did you expect to see?

I expected to see output similar to task manager where every process has a associated username.

### What did you see instead?

```
+------+----------+------+-------------------------------------------------------------------------+
| pid  | username | uid  | path                                                                    |
+------+----------+------+-------------------------------------------------------------------------+
| 0    |          | -1   |                                                                         |
| 4    |          | 0    |                                                                         |
| 300  |          | 544  |                                                                         |
| 376  |          | 544  | C:\Windows\system32\csrss.exe                                           |
...
| 632  |          | -1   | C:\Windows\system32\svchost.exe                                         |
| 700  |          | 0    | C:\Windows\system32\svchost.exe                                         |
| 752  |          | 0    | C:\Windows\System32\svchost.exe                                         |
| 816  |          | 0    | C:\Windows\system32\LogonUI.exe                                         |
| 856  | SYSTEM   | 18   | C:\Windows\System32\svchost.exe                                         |
| 880  |          | -1   | C:\Windows\system32\svchost.exe                                         |
| 912  |          | 0    | C:\Windows\system32\svchost.exe                                         |
| 976  |          | 544  | C:\Windows\system32\svchost.exe                                         |
| 900  |          | -1   | C:\Windows\system32\svchost.exe                                         |
| 1092 |          | 0    | C:\Windows\System32\spoolsv.exe                                         |
| 1128 |          | 0    | C:\Windows\system32\svchost.exe                                         |
| 1348 |          | 0    | C:\Windows\System32\svchost.exe                                         |
...
| 908  |          | -1   | C:\Windows\system32\svchost.exe                                         |
| 1200 |          | 0    | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 1792 |          | -1   | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 2420 |          | 0    | C:\Program Files (x86)\Citrix\XenTools\XenDpriv.exe                     |
| 2712 |          | -1   | C:\Windows\System32\svchost.exe                                         |
| 2524 |          | 0    | C:\Windows\system32\csrss.exe                                           |
| 2588 |          | 544  | C:\Windows\system32\winlogon.exe                                        |
| 3032 |          | -1   | C:\Windows\system32\sppsvc.exe                                          |
| 3056 |          | 0    | C:\Windows\system32\SearchIndexer.exe                                   |
| 1904 |          | 0    | C:\Windows\system32\taskhost.exe                                        |
| 1948 | mike     | 1106 | C:\Windows\system32\rdpclip.exe                                         |
| 2168 | mike     | 1106 | C:\Windows\system32\Dwm.exe                                             |
...
| 2564 |          | -1   | C:\Windows\system32\svchost.exe                                         |
| 4056 |          | 0    | C:\Windows\Microsoft.Net\Framework64\v3.0\WPF\PresentationFontCache.exe |
| 3600 |          | 0    | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 3976 |          | 544  | C:\Windows\system32\conhost.exe                                         |
| 3024 |          | 544  | C:\Windows\system32\csrss.exe                                           |
| 2240 |          | 544  | C:\Windows\system32\winlogon.exe                                        |
| 2880 |          | -1   | C:\Windows\system32\taskhost.exe                                        |
| 3436 |          | 0    | C:\Windows\system32\rdpclip.exe                                         |
| 3188 |          | 0    | C:\Windows\system32\Dwm.exe                                             |
| 844  |          | 0    | C:\Windows\Explorer.EXE                                                 |
| 2256 |          | 0    | C:\Windows\system32\taskhost.exe                                        |
| 3204 |          | 544  | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 4868 |          | 544  | C:\Windows\system32\conhost.exe                                         |
| 1656 |          | 544  | C:\Windows\system32\mmc.exe                                             |
| 1552 |          | -1   | C:\Windows\system32\UI0Detect.exe                                       |
| 2664 |          | 0    | C:\Windows\system32\mmc.exe                                             |
| 4780 | mike     | 1106 | C:\Windows\explorer.exe                                                 |
| 1560 |          | 544  | C:\Windows\System32\perfmon.exe                                         |
| 4804 |          | 544  | C:\Program Files\Internet Explorer\iexplore.exe                         |
| 832  |          | 544  | C:\Program Files (x86)\Internet Explorer\IEXPLORE.EXE                   |
| 4692 |          | -1   | C:\Windows\system32\NOTEPAD.EXE                                         |
| 4264 |          | 0    | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 5012 |          | 544  | C:\Windows\system32\conhost.exe                                         |
| 2204 |          | 544  | C:\Windows\system32\mmc.exe                                             |
| 4856 |          | 544  | C:\programdata\osquery\osqueryi.exe                                     |
| 1784 |          | 544  | C:\Windows\system32\taskmgr.exe                                         |
| 4652 |          | 544  | C:\ProgramData\osquery\osqueryd\osqueryd.exe                            |
| 2296 |          | 544  | C:\ProgramData\osquery\osqueryd\osqueryd.exe                            |
| 3628 |          | 544  | C:\Windows\system32\conhost.exe                                         |
| 3944 |          | 544  | C:\Windows\regedit.exe                                                  |
+------+----------+------+-------------------------------------------------------------------------+
```

The rows with empty usernames is because those uid's don't exist in the `users` table which is partially the problem. Additionally, I think some of the uid's are RIDs instead of the uid of the account that spawned the process. For example, process 4264 is actually a PowerShell console in an elevated command prompt spawned by the user "mike" session running as local administrator. You can see in the screenshot below that the same process is attributed to the original session owner user name.

![image](https://user-images.githubusercontent.com/7902735/39557649-fea75dc2-4e56-11e8-9294-f2981bdb67e4.png)

uid 544 is the [known RID](https://support.microsoft.com/en-us/help/243330/well-known-security-identifiers-in-windows-operating-systems) of the local Administrators group. So because builtin accounts are not in the users table we can't map what builtin account a process is associated with, but more importantly there doesn't appear to be a way to get the original user account that spawned the elevated prompt. This makes osquery effectively unable to attribute an elevated process to a specific user; even across multiple sessions.

**Comment by Rich5** [comment]
What can I do to help out with this?

**Comment by muffins** [comment]
Hey @Rich5, sorry for no response on this, I just haven't had a chance to tear into it. We have a few other issues noting some shortcomings of the users table, so it looks like there might be some re-architecting that needs doing there. If you're comfortable with C++ and you're interested in digging into the fix I'd suspect there's some changes that need done to the [users table](https://github.com/facebook/osquery/blob/master/osquery/tables/system/windows/users.cpp#L166), and _maybe_ even the processes table, but I'm not 100%. Let me know if that's something you're feeling up to taking this on! If not I'll bump this up in my priority as this is a relatively big issue :)

**Comment by Rich5** [comment]
No worries. Just wanted to offer help if I can. I'll take a look at the users and processes table to see if I can triage. I know my way around code, but I'm not the best C++ dev so I'm not sure I'm the one to ultimately implement the fix. If it's something I can do myself I'll get a PR together though.

**Comment by Rich5** [comment]
After looking around a bit I think a quick fix might be just changing the parameters to GetTokenInformation. You can see [here](https://github.com/facebook/osquery/blob/master/osquery/tables/system/windows/processes.cpp#L238 ) that TokenOwner is used instead of TokenUser.  TokenOwner appears to return the group rather than the user which may be why we get 544 and 0 for uid for elevated processes. This [answer ](https://stackoverflow.com/questions/2686096/c-get-username-from-process)on SO seems to confirm this method; as well as the MS docs on [TOKEN_INFORMATION_CLASS](https://msdn.microsoft.com/en-us/library/windows/desktop/aa379626(v=vs.85).aspx). I haven't tested this yet because I need to setup a osquery dev environment, but if this works we shouldn't need any edits to the users table for the near term. For even more useful information adding a column to show whether the [token is elevated](https://msdn.microsoft.com/en-us/library/windows/desktop/bb530718%28v=vs.85%29.aspx) would be great, and looks like just another call to GetTokenInformation. Then we can show process owners and whether they are using elevated processes.

**Comment by muffins** [comment]
@Rich5 awesome triage! Thanks for digging into this! Yeah I don't think there was any _specific_ reason for pulling the group token information as opposed to the user, but @yying could correct me if I'm wrong here. I'd say lets move forward with the switch to `TokenUser` as opposed to `TokenOwner` if it's getting us more accurate information.

As for extending the schema with an `is_elevated_token` style column, that's a great idea. [We have a concept of an extended](https://github.com/facebook/osquery/blob/master/specs/groups.table#L8-L11) schema that allows one to have additional columns for specific platforms, this could be used for the `is_elevated_token` information

**Comment by Rich5** [comment]
I'm standing up a dev environment so I can test this out. I'll keep you posted.

**Comment by Rich5** [comment]
I can confirm that using TokenUser works!

Using TokenOwner
```
+------+-----------------+------+-------------------------------------------------------------------------+
| pid  | username        | uid  | path                                                                    |
+------+-----------------+------+-------------------------------------------------------------------------+
| 0    |                 | -1   |                                                                         |
| 4    |                 | 0    |                                                                         |
| 300  |                 | 544  |                                                                         |
| 376  |                 | 544  | C:\Windows\system32\csrss.exe                                           |
| 452  |                 | 544  | C:\Windows\system32\wininit.exe                                         |
| 504  |                 | 544  | C:\Windows\system32\services.exe                                        |
| 520  |                 | 544  | C:\Windows\system32\lsass.exe                                           |
| 528  |                 | 544  | C:\Windows\system32\lsm.exe                                             |
| 632  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         |
| 700  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         |
| 752  | LOCAL SERVICE   | 19   | C:\Windows\System32\svchost.exe                                         |
| 856  | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         |
| 880  | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         |
| 912  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         |
| 976  |                 | 544  | C:\Windows\system32\svchost.exe                                         |
| 900  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         |
| 1092 | SYSTEM          | 18   | C:\Windows\System32\spoolsv.exe                                         |
| 1128 | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         |
| 1348 | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         |
| 1580 |                 | 544  | C:\Program Files (x86)\TeamViewer\TeamViewer_Service.exe                |
| 1624 |                 | 544  | C:\ProgramData\ZeroTier\One\zerotier-one_x64.exe                        |
| 1820 |                 | 544  | C:\Program Files\Amazon\Ec2ConfigService\Ec2Config.exe                  |
| 1868 |                 | 544  | C:\Program Files (x86)\Citrix\XenTools\XenGuestAgent.exe                |
| 908  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         |
| 1200 | SYSTEM          | 18   | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 1792 | NETWORK SERVICE | 20   | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 2712 | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         |
| 2524 |                 | 544  | C:\Windows\system32\csrss.exe                                           |
| 2588 |                 | 544  | C:\Windows\system32\winlogon.exe                                        |
| 3032 | NETWORK SERVICE | 20   | C:\Windows\system32\sppsvc.exe                                          |
| 3056 | SYSTEM          | 18   | C:\Windows\system32\SearchIndexer.exe                                   |
| 1904 | mike            | 1106 | C:\Windows\system32\taskhost.exe                                        |
| 2168 | mike            | 1106 | C:\Windows\system32\Dwm.exe                                             |
| 2316 | mike            | 1106 | C:\Program Files (x86)\ZeroTier\One\ZeroTier One.exe                    |
| 2564 | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         |
| 4056 | LOCAL SERVICE   | 19   | C:\Windows\Microsoft.Net\Framework64\v3.0\WPF\PresentationFontCache.exe |
| 3600 |                 | 544  | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 3976 |                 | 544  | C:\Windows\system32\conhost.exe                                         |
| 3024 |                 | 544  | C:\Windows\system32\csrss.exe                                           |
| 2240 |                 | 544  | C:\Windows\system32\winlogon.exe                                        |
| 2880 | jsmith          | 1109 | C:\Windows\system32\taskhost.exe                                        |
| 3436 | jsmith          | 1109 | C:\Windows\system32\rdpclip.exe                                         |
| 3188 | jsmith          | 1109 | C:\Windows\system32\Dwm.exe                                             |
| 844  | jsmith          | 1109 | C:\Windows\Explorer.EXE                                                 |
| 2256 |                 | 544  | C:\Windows\system32\taskhost.exe                                        |
| 3204 |                 | 544  | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 4868 |                 | 544  | C:\Windows\system32\conhost.exe                                         |
| 1552 | SYSTEM          | 18   | C:\Windows\system32\UI0Detect.exe                                       |
| 2664 |                 | 544  | C:\Windows\system32\mmc.exe                                             |
| 4780 | mike            | 1106 | C:\Windows\explorer.exe                                                 |
| 4804 |                 | 544  | C:\Program Files\Internet Explorer\iexplore.exe                         |
| 832  |                 | 544  | C:\Program Files (x86)\Internet Explorer\IEXPLORE.EXE                   |
| 4692 | jsmith          | 1109 | C:\Windows\system32\NOTEPAD.EXE                                         |
| 4264 |                 | 544  | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 5012 |                 | 544  | C:\Windows\system32\conhost.exe                                         |
| 1784 |                 | 544  | C:\Windows\system32\taskmgr.exe                                         |
| 5868 | mike            | 1106 | c:\program files (x86)\teamviewer\TeamViewer.exe                        |
| 3764 |                 | 544  | C:\Program Files (x86)\TeamViewer\tv_w32.exe                            |
| 5980 |                 | 544  | C:\Program Files (x86)\TeamViewer\tv_x64.exe                            |
| 3208 |                 | 544  | C:\ProgramData\osquery\osqueryd\osqueryd.exe                            |
| 624  |                 | 544  | C:\ProgramData\osquery\osqueryd\osqueryd.exe                            |
| 5352 |                 | 544  | C:\Windows\system32\conhost.exe                                         |
| 1512 |                 | 544  | C:\Windows\system32\csrss.exe                                           |
| 2436 |                 | 544  | C:\Windows\system32\winlogon.exe                                        |
| 3888 |                 | 544  | C:\Windows\system32\LogonUI.exe                                         |
| 4216 | mike            | 1106 | C:\Windows\system32\rdpclip.exe                                         |
| 4492 |                 | 544  | C:\Program Files (x86)\Citrix\XenTools\XenDpriv.exe                     |
| 2276 |                 | 544  | C:\Windows\system32\cmd.exe                                             |
| 2532 |                 | 544  | C:\Windows\system32\conhost.exe                                         |
| 4828 |                 | 544  | c:\Users\mike\Downloads\SysinternalsSuite\PsExec.exe                    |
| 5792 |                 | 544  | C:\Windows\PSEXESVC.exe                                                 |
| 5680 |                 | 544  | C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe               |
| 5732 |                 | 544  | C:\Windows\system32\conhost.exe                                         |
| 5256 | LOCAL SERVICE   | 19   | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 1376 |                 | 544  | c:\programdata\osquery\osqueryi-old.exe                                 |
+------+-----------------+------+-------------------------------------------------------------------------+

```

using TokenUser

```
+------+-----------------+------+-------------------------------------------------------------------------+
| pid  | username        | uid  | path                                                                    |
+------+-----------------+------+-------------------------------------------------------------------------+
| 0    |                 | -1   |                                                                         |
| 4    |                 | 0    |                                                                         |
| 300  | SYSTEM          | 18   |                                                                         |
| 376  | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           |
| 452  | SYSTEM          | 18   | C:\Windows\system32\wininit.exe                                         |
| 504  | SYSTEM          | 18   | C:\Windows\system32\services.exe                                        |
| 520  | SYSTEM          | 18   | C:\Windows\system32\lsass.exe                                           |
| 528  | SYSTEM          | 18   | C:\Windows\system32\lsm.exe                                             |
| 632  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         |
| 700  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         |
| 752  | LOCAL SERVICE   | 19   | C:\Windows\System32\svchost.exe                                         |
| 856  | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         |
| 880  | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         |
| 912  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         |
| 976  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         |
| 900  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         |
| 1092 | SYSTEM          | 18   | C:\Windows\System32\spoolsv.exe                                         |
| 1128 | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         |
| 1348 | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         |
| 1580 | SYSTEM          | 18   | C:\Program Files (x86)\TeamViewer\TeamViewer_Service.exe                |
| 1624 | SYSTEM          | 18   | C:\ProgramData\ZeroTier\One\zerotier-one_x64.exe                        |
| 1820 | SYSTEM          | 18   | C:\Program Files\Amazon\Ec2ConfigService\Ec2Config.exe                  |
| 1868 | SYSTEM          | 18   | C:\Program Files (x86)\Citrix\XenTools\XenGuestAgent.exe                |
| 908  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         |
| 1200 | SYSTEM          | 18   | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 1792 | NETWORK SERVICE | 20   | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 2712 | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         |
| 2524 | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           |
| 2588 | SYSTEM          | 18   | C:\Windows\system32\winlogon.exe                                        |
| 3032 | NETWORK SERVICE | 20   | C:\Windows\system32\sppsvc.exe                                          |
| 3056 | SYSTEM          | 18   | C:\Windows\system32\SearchIndexer.exe                                   |
| 1904 | mike            | 1106 | C:\Windows\system32\taskhost.exe                                        |
| 2168 | mike            | 1106 | C:\Windows\system32\Dwm.exe                                             |
| 2316 | mike            | 1106 | C:\Program Files (x86)\ZeroTier\One\ZeroTier One.exe                    |
| 2564 | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         |
| 4056 | LOCAL SERVICE   | 19   | C:\Windows\Microsoft.Net\Framework64\v3.0\WPF\PresentationFontCache.exe |
| 3600 | mike            | 1106 | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 3976 | mike            | 1106 | C:\Windows\system32\conhost.exe                                         |
| 3024 | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           |
| 2240 | SYSTEM          | 18   | C:\Windows\system32\winlogon.exe                                        |
| 2880 | jsmith          | 1109 | C:\Windows\system32\taskhost.exe                                        |
| 3436 | jsmith          | 1109 | C:\Windows\system32\rdpclip.exe                                         |
| 3188 | jsmith          | 1109 | C:\Windows\system32\Dwm.exe                                             |
| 844  | jsmith          | 1109 | C:\Windows\Explorer.EXE                                                 |
| 2256 | jsmith          | 1109 | C:\Windows\system32\taskhost.exe                                        |
| 3204 | jsmith          | 1109 | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 4868 | jsmith          | 1109 | C:\Windows\system32\conhost.exe                                         |
| 1552 | SYSTEM          | 18   | C:\Windows\system32\UI0Detect.exe                                       |
| 2664 | mike            | 1106 | C:\Windows\system32\mmc.exe                                             |
| 4780 | mike            | 1106 | C:\Windows\explorer.exe                                                 |
| 4804 | mike            | 1106 | C:\Program Files\Internet Explorer\iexplore.exe                         |
| 832  | mike            | 1106 | C:\Program Files (x86)\Internet Explorer\IEXPLORE.EXE                   |
| 4692 | jsmith          | 1109 | C:\Windows\system32\NOTEPAD.EXE                                         |
| 4264 | mike            | 1106 | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               |
| 5012 | mike            | 1106 | C:\Windows\system32\conhost.exe                                         |
| 1784 | mike            | 1106 | C:\Windows\system32\taskmgr.exe                                         |
| 5868 | mike            | 1106 | c:\program files (x86)\teamviewer\TeamViewer.exe                        |
| 3764 | SYSTEM          | 18   | C:\Program Files (x86)\TeamViewer\tv_w32.exe                            |
| 5980 | SYSTEM          | 18   | C:\Program Files (x86)\TeamViewer\tv_x64.exe                            |
| 3208 | SYSTEM          | 18   | C:\ProgramData\osquery\osqueryd\osqueryd.exe                            |
| 624  | SYSTEM          | 18   | C:\ProgramData\osquery\osqueryd\osqueryd.exe                            |
| 5352 | SYSTEM          | 18   | C:\Windows\system32\conhost.exe                                         |
| 1512 | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           |
| 2436 | SYSTEM          | 18   | C:\Windows\system32\winlogon.exe                                        |
| 3888 | SYSTEM          | 18   | C:\Windows\system32\LogonUI.exe                                         |
| 4216 | mike            | 1106 | C:\Windows\system32\rdpclip.exe                                         |
| 4492 | SYSTEM          | 18   | C:\Program Files (x86)\Citrix\XenTools\XenDpriv.exe                     |
| 2276 | mike            | 1106 | C:\Windows\system32\cmd.exe                                             |
| 2532 | mike            | 1106 | C:\Windows\system32\conhost.exe                                         |
| 4828 | mike            | 1106 | c:\Users\mike\Downloads\SysinternalsSuite\PsExec.exe                    |
| 5792 | SYSTEM          | 18   | C:\Windows\PSEXESVC.exe                                                 |
| 5680 | SYSTEM          | 18   | C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe               |
| 5732 | SYSTEM          | 18   | C:\Windows\system32\conhost.exe                                         |
| 5256 | LOCAL SERVICE   | 19   | C:\Windows\system32\wbem\wmiprvse.exe                                   |
| 5936 | SYSTEM          | 18   | c:\programdata\osquery\osqueryi.exe                                     |
| 3224 | mike            | 1106 | C:\Windows\system32\DllHost.exe                                         |
+------+-----------------+------+-------------------------------------------------------------------------+
```

I'll start working on the `is_elevated_token` column and get a PR together.

**Comment by muffins** [comment]
That's awesome, thanks @Rich5!

**Comment by Rich5** [comment]
And there we go...

```
+------+-----------------+------+-------------------------------------------------------------------------+-------------------+
| pid  | username        | uid  | path                                                                    | is_elevated_token |
+------+-----------------+------+-------------------------------------------------------------------------+-------------------+
| 0    |                 | -1   |                                                                         |                   |
| 4    |                 | 0    |                                                                         |                   |
| 300  | SYSTEM          | 18   |                                                                         | 1                 |
| 376  | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           | 1                 |
| 452  | SYSTEM          | 18   | C:\Windows\system32\wininit.exe                                         | 1                 |
| 504  | SYSTEM          | 18   | C:\Windows\system32\services.exe                                        | 1                 |
| 520  | SYSTEM          | 18   | C:\Windows\system32\lsass.exe                                           | 1                 |
| 528  | SYSTEM          | 18   | C:\Windows\system32\lsm.exe                                             | 1                 |
| 632  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 700  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 752  | LOCAL SERVICE   | 19   | C:\Windows\System32\svchost.exe                                         | 1                 |
| 856  | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         | 1                 |
| 880  | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 912  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 976  | SYSTEM          | 18   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 900  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 1092 | SYSTEM          | 18   | C:\Windows\System32\spoolsv.exe                                         | 1                 |
| 1128 | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 1348 | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         | 1                 |
| 1580 | SYSTEM          | 18   | C:\Program Files (x86)\TeamViewer\TeamViewer_Service.exe                | 1                 |
| 1624 | SYSTEM          | 18   | C:\ProgramData\ZeroTier\One\zerotier-one_x64.exe                        | 1                 |
| 1820 | SYSTEM          | 18   | C:\Program Files\Amazon\Ec2ConfigService\Ec2Config.exe                  | 1                 |
| 1868 | SYSTEM          | 18   | C:\Program Files (x86)\Citrix\XenTools\XenGuestAgent.exe                | 1                 |
| 908  | NETWORK SERVICE | 20   | C:\Windows\system32\svchost.exe                                         | 1                 |
| 1200 | SYSTEM          | 18   | C:\Windows\system32\wbem\wmiprvse.exe                                   | 1                 |
| 1792 | NETWORK SERVICE | 20   | C:\Windows\system32\wbem\wmiprvse.exe                                   | 1                 |
| 2712 | SYSTEM          | 18   | C:\Windows\System32\svchost.exe                                         | 1                 |
| 2524 | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           | 1                 |
| 2588 | SYSTEM          | 18   | C:\Windows\system32\winlogon.exe                                        | 1                 |
| 3032 | NETWORK SERVICE | 20   | C:\Windows\system32\sppsvc.exe                                          | 1                 |
| 3056 | SYSTEM          | 18   | C:\Windows\system32\SearchIndexer.exe                                   | 1                 |
| 1904 | mike            | 1106 | C:\Windows\system32\taskhost.exe                                        | 0                 |
| 2168 | mike            | 1106 | C:\Windows\system32\Dwm.exe                                             | 0                 |
| 2316 | mike            | 1106 | C:\Program Files (x86)\ZeroTier\One\ZeroTier One.exe                    | 0                 |
| 2564 | LOCAL SERVICE   | 19   | C:\Windows\system32\svchost.exe                                         | 0                 |
| 4056 | LOCAL SERVICE   | 19   | C:\Windows\Microsoft.Net\Framework64\v3.0\WPF\PresentationFontCache.exe | 1                 |
| 3600 | mike            | 1106 | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               | 1                 |
| 3976 | mike            | 1106 | C:\Windows\system32\conhost.exe                                         | 1                 |
| 3024 | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           | 1                 |
| 2240 | SYSTEM          | 18   | C:\Windows\system32\winlogon.exe                                        | 1                 |
| 2880 | jsmith          | 1109 | C:\Windows\system32\taskhost.exe                                        | 0                 |
| 3436 | jsmith          | 1109 | C:\Windows\system32\rdpclip.exe                                         | 0                 |
| 3188 | jsmith          | 1109 | C:\Windows\system32\Dwm.exe                                             | 0                 |
| 844  | jsmith          | 1109 | C:\Windows\Explorer.EXE                                                 | 0                 |
| 2256 | jsmith          | 1109 | C:\Windows\system32\taskhost.exe                                        | 1                 |
| 3204 | jsmith          | 1109 | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               | 1                 |
| 4868 | jsmith          | 1109 | C:\Windows\system32\conhost.exe                                         | 1                 |
| 1552 | SYSTEM          | 18   | C:\Windows\system32\UI0Detect.exe                                       | 1                 |
| 2664 | mike            | 1106 | C:\Windows\system32\mmc.exe                                             | 1                 |
| 4780 | mike            | 1106 | C:\Windows\explorer.exe                                                 | 0                 |
| 4804 | mike            | 1106 | C:\Program Files\Internet Explorer\iexplore.exe                         | 1                 |
| 832  | mike            | 1106 | C:\Program Files (x86)\Internet Explorer\IEXPLORE.EXE                   | 1                 |
| 4692 | jsmith          | 1109 | C:\Windows\system32\NOTEPAD.EXE                                         | 0                 |
| 4264 | mike            | 1106 | C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe               | 1                 |
| 5012 | mike            | 1106 | C:\Windows\system32\conhost.exe                                         | 1                 |
| 1784 | mike            | 1106 | C:\Windows\system32\taskmgr.exe                                         | 1                 |
| 5868 | mike            | 1106 | c:\program files (x86)\teamviewer\TeamViewer.exe                        | 0                 |
| 3764 | SYSTEM          | 18   | C:\Program Files (x86)\TeamViewer\tv_w32.exe                            | 1                 |
| 5980 | SYSTEM          | 18   | C:\Program Files (x86)\TeamViewer\tv_x64.exe                            | 1                 |
| 1512 | SYSTEM          | 18   | C:\Windows\system32\csrss.exe                                           | 1                 |
| 2436 | SYSTEM          | 18   | C:\Windows\system32\winlogon.exe                                        | 1                 |
| 3888 | SYSTEM          | 18   | C:\Windows\system32\LogonUI.exe                                         | 1                 |
| 4216 | mike            | 1106 | C:\Windows\system32\rdpclip.exe                                         | 0                 |
| 4492 | SYSTEM          | 18   | C:\Program Files (x86)\Citrix\XenTools\XenDpriv.exe                     | 1                 |
| 2276 | mike            | 1106 | C:\Windows\system32\cmd.exe                                             | 1                 |
| 2532 | mike            | 1106 | C:\Windows\system32\conhost.exe                                         | 1                 |
| 4828 | mike            | 1106 | c:\Users\mike\Downloads\SysinternalsSuite\PsExec.exe                    | 1                 |
| 5792 | SYSTEM          | 18   | C:\Windows\PSEXESVC.exe                                                 | 1                 |
| 5680 | SYSTEM          | 18   | C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe               | 1                 |
| 5732 | SYSTEM          | 18   | C:\Windows\system32\conhost.exe                                         | 1                 |
| 4436 | SYSTEM          | 18   | C:\ProgramData\osquery\osqueryd\osqueryd.exe                            | 1                 |
| 3820 | SYSTEM          | 18   | C:\Windows\system32\conhost.exe                                         | 1                 |
| 4832 | LOCAL SERVICE   | 19   | C:\Windows\system32\wbem\wmiprvse.exe                                   | 1                 |
| 2988 |                 | 0    |                                                                         |                   |
| 1324 | SYSTEM          | 18   | c:\program files (x86)\teamviewer\TeamViewer_Desktop.exe                | 1                 |
| 6012 | SYSTEM          | 18   | C:\Windows\system32\taskeng.exe                                         | 1                 |
| 5368 | SYSTEM          | 18   | c:\programdata\osquery\osqueryi.exe                                     | 1                 |
+------+-----------------+------+-------------------------------------------------------------------------+-------------------+
```

**Comment by muffins** [comment]
Fixed in #4369
